BuildLibs/buildmap.py

```diff
@@ -1,153 +1,190 @@
-from struct import *
-from BuildLibs import *
-from BuildLibs import games
+import binascii
+import copy
+import random
+import struct
 from pathlib import Path
+from typing import OrderedDict, Union
+
+from BuildLibs import crc32, error, games, info, swapobjkey, trace, warning, FancyPacker
+
 
 class CStat:
+
     def __init__(self, cstat):
         self.blocking = bool(cstat & 1)
         self.facing = cstat & 0x30
         self.onesided = cstat & 0x40
         self.blockingHitscan = bool(cstat & 0x800)
         self.invisible = bool(cstat & 0x8000)
 
+
 class Sector:
-    def __init__(self, **kargs):
-        self.__dict__ = dict(**kargs)
-        self.nearbySectors:set
-        self.walls:list
-        self.shapes:list
+
+    def __init__(self, **kwargs):
+
+        self.wall: int = kwargs.get('wall', 0)
+        self.wallnum: int = kwargs.get('wallnum', 0)
+        self.ceilingz: int = kwargs.get('ceilingz', 0)
+        self.floorz: int = kwargs.get('floorz', 0)
+        self.ceilingstat: int = kwargs.get('ceilingstat', 0)
+        self.floorstat: int = kwargs.get('floorstat', 0)
+        self.ceilingpicnum: int = kwargs.get('ceilingpicnum', 0)
+        self.ceilingheinum: int = kwargs.get('ceilingheinum', 0)
+        self.ceilingshade: int = kwargs.get('ceilingshade', 0)
+        self.ceiling_palette: int = kwargs.get('ceiling_palette', 0)
+        self.ceiling_texcoords: list = kwargs.get('ceiling_texcoords', 0)
+        self.floorpicnum: int = kwargs.get('floorpicnum', 0)
+        self.floorheinum: int = kwargs.get('floorheinum', 0)
+        self.floorshade: int = kwargs.get('floorshade', 0)
+        self.floor_palette: int = kwargs.get('floor_palette', 0)
+        self.floor_texcoords: list = kwargs.get('floor_texcoords', [0, 0])
+        self.visibility: int = kwargs.get('visibility', 0)
+        self.filler: int = kwargs.get('filler', 0)
+        self.lowtag: int = kwargs.get('lowtag', 0)
+        self.hightag: int = kwargs.get('hightag', 0)
+        self.extra: int = kwargs.get('extra', -1)
+        self.length: int = kwargs.get('length', 0)
+
+        # TODO: texcoords
+        self.ceilingxpanning: int = kwargs.get('ceilingxpanning', 0)
+        self.floorxpanning: int = kwargs.get('floorxpanning', 0)
+        self.ceilingypanning: int = kwargs.get('ceilingypanning', 0)
+        self.floorypanning: int = kwargs.get('floorypanning', 0)
+
+        self.walls: Union[list, None] = kwargs.get('walls', None)
+        self.nearbySectors: Union[set, None] = kwargs.get('nearbySectors', None)
+        self.shapes: Union[list, None] = kwargs.get('shapes', None)
+
+
+class Wall:
+
+    def __init__(self, **kwargs):
+        self.pos: list = kwargs.get('pos', [0, 0])
+        self.point2: int = kwargs.get('point2', 0)
```

Comment:
maybe rename `point2` to `next_wall_in_shape`, and rename `nextwall` to `other_sector_wall`?
```suggestion
        self.next_wall_in_shape: int = kwargs.get('next_wall_in_shape', 0) # according to moddingwiki this is called point2
```